agdb/tests/db_user_value_test.rs

```diff
@@ -35,7 +37,7 @@ impl From<Status> for DbValue {
 impl TryFrom<DbValue> for Status {
     type Error = DbError;
 
-    fn try_from(value: DbValue) -> Result<Self, Self::Error> {
+    fn try_from(value: DbValue) -> std::result::Result<Self, Self::Error> {
```

Comment:
Note that this isn't enough if somebody overrides `std`. Same with stuff like `Ok` or `DbValue` too. Thus a leading `::` is necessary too (known as global paths)

Reply:
Oh this actually isn't a macro, so pretend that this comment was below

Reply:
Hm, interesting. It's a little different to the ambiguity of imports thugh. If you override `std` then you actually want it to be overriden so should a library author use `::std` to prevent that?

Reply:
That's an interesting point, though I haven't really seen it used. It's definitely possible to do that elsewhere (with patches in cargo, not really with std) but it might be unexpected here.

Reply:
Somebody could also do this anyway by passing `--extern` with a custom `std`, afaik.

Reply:
> but it might be unexpected here.

Looking at some crates, namely `serde` and `axum` I can see the mix. `serde` is not using `::` prefix so patching `std` in code is possible but `axum` is enforcing `::std` so doing it in code is not possible. The results of patching `std` in code with those two dependencies would no doubt be "interesting". I would assume when using `--extern` the patched `std` is used everywhere regardless.

So my conclusion is that using or not using `::` makes little practical difference because nobody should rely on it in code because dependencies will inevitably use a mix of both. Restricing the patched `std` to specific code similarly to the use of `unsafe` is advisable. And when there is a need for patched `std` across the board using `--extern` should solve it globally and then the usage (or not) of `::` does not matter.

Thank you for this though. It really reminded me of the goold old C++ days where stuff like this is absolutely common. Thank god for Rust where it's find only in the fringes. :-)